Updated package dependencies and added Markdown Filename Checks

NebraskaCoder opened the pull request:
Updated Node package dependencies and adding checking for dots in Markdown filenames on both precommit and a GitHub Workflow.

ressonix commented:
We can't action use the term "RHEL" on the site, which is why our English site uses "Enterprise Linux". Other than that, I don't see anything else that would absolutely need changing like that.

g3vxy commented:
I changed all the "RHEL"s to "Enterprise Linux". There will be no difference in the meaning of the sentence.

NebraskaCoder committed:
Update package versions

NebraskaCoder committed:
chore: Update lint-staged.config.js to include precommit check for .md and .mdx filenames

NebraskaCoder committed:
feat: Add GitHub workflow to validate Markdown filenames

NebraskaCoder committed:
feat: Add validation to skip Storybook Markdown filenames

NebraskaCoder committed:
feat: Refactor Markdown filename validation workflow

The code changes in this commit refactor the GitHub workflow for validating Markdown filenames. The previous implementation was updated to skip Storybook Markdown filenames and properly handle invalid filenames. This change improves the efficiency and accuracy of the validation process.

Note: This commit message follows the established conventions in the repository.
